crates/solver-service/src/apis/order.rs

```diff
@@ -0,0 +1,140 @@
+use uuid::Uuid;
+
+/// Handles GET /order/{id} requests.
+///
+/// This endpoint retrieves order details by ID, providing status information
+/// and execution details for cross-chain intent orders.
+pub async fn get_order_by_id(
+	Path(id): Path<String>,
+	_solver: &SolverEngine,
+) -> Result<GetOrderResponse, GetOrderError> {
+	info!("Retrieving order with ID: {}", id);
+
+	let order = process_order_request(&id, _solver).await?;
+
+	Ok(GetOrderResponse { order })
+}
+
+/// Processes an order retrieval request.
+async fn process_order_request(
+	order_id: &str,
+	solver: &SolverEngine,
+) -> Result<OrderResponse, GetOrderError> {
+	// Validate order ID format
+	validate_order_id(order_id)?;
+
+	// Try to retrieve the order from storage
+	match solver
+		.storage()
+		.retrieve::<solver_types::Order>("orders", order_id)
+		.await
+	{
+		Ok(order) => {
+			// Order found in storage, convert to OrderResponse
+			convert_order_to_response(order).await
+		}
+		Err(solver_storage::StorageError::NotFound) => {
+			// Order not found in storage
+			Err(GetOrderError::NotFound(format!(
+				"Order not found: {}",
+				order_id
+			)))
+		}
+		Err(e) => {
+			// Other storage error
+			Err(GetOrderError::Internal(format!("Storage error: {}", e)))
+		}
+	}
+}
+
+/// Validates the order ID format.
+fn validate_order_id(order_id: &str) -> Result<(), GetOrderError> {
+	// Check if it's a valid UUID format
+	if Uuid::parse_str(order_id).is_err() {
+		return Err(GetOrderError::InvalidId(format!(
+			"Order ID must be a valid UUID: {}",
+			order_id
+		)));
+	}
+
+	Ok(())
+}
+
+/// Converts a storage Order to an API OrderResponse.
+async fn convert_order_to_response(
+	order: solver_types::Order,
+) -> Result<OrderResponse, GetOrderError> {
+	// Extract data from the order's JSON data field
+	// This assumes the order.data contains the necessary fields for the API response
+	let input_amount = order
+		.data
+		.get("inputAmount")
+		.and_then(|v| serde_json::from_value::<AssetAmount>(v.clone()).ok())
+		.unwrap_or_else(|| AssetAmount {
+			asset: "0x0000000000000000000000000000000000000000".to_string(),
+			amount: alloy_primitives::U256::ZERO,
+		});
+
+	let output_amount = order
+		.data
+		.get("outputAmount")
+		.and_then(|v| serde_json::from_value::<AssetAmount>(v.clone()).ok())
+		.unwrap_or_else(|| AssetAmount {
+			asset: "0x0000000000000000000000000000000000000000".to_string(),
+			amount: alloy_primitives::U256::ZERO,
+		});
+
+	let settlement_type = order
+		.data
+		.get("settlementType")
+		.and_then(|v| serde_json::from_value::<SettlementType>(v.clone()).ok())
+		.unwrap_or(SettlementType::Escrow);
+
+	let settlement_data = order
+		.data
+		.get("settlementData")
+		.cloned()
+		.unwrap_or_else(|| serde_json::json!({}));
+
+	// For now, assume all orders are pending unless we have more status tracking
+	// In a real implementation, you would query additional storage to determine the actual status
+	let status = DetailedIntentStatus::Pending;
```

Comment:
The main reason for this endpoint is status tracking, let's not default on one status